### PR TITLE
Update to APA provost

### PR DESCRIPTION
**Pull request by fharnois**
Update to correct some inconsistencies regarding the use of volume field, to add some document types (e.g. Patent) and to add genre between [ ] for some documents. All those changes were made in respect with the Provost Guidelines in « Normes de présentation d'un travail de recherche » 4th edition.

### CODE REVIEW

File: apa-fr-provost.csl [hunk 1]
```diff
@@ -174,36 +174,60 @@
...
         <group prefix=" (" suffix=")" delimiter=" ">
-          <text variable="genre"/>
-          <text variable="number" prefix="No. "/>
+          <choose>
+            <if variable="number" match="any">
+              <text value="Rapport" suffix=" "/>
+              <text variable="number" prefix="No. "/>
+            </if>
+          </choose>
         </group>
```
Comment (rmzelle) [inline apa-fr-provost.csl]: @fharnois, you should be able to change

```xml
        <group prefix=" (" suffix=")" delimiter=" ">
          <choose>
            <if variable="number" match="any">
              <text value="Rapport" suffix=" "/>
              <text variable="number" prefix="No. "/>
            </if>
          </choose>
        </group>
```

to the simpler

```xml
        <group prefix=" (" suffix=")" delimiter=" ">
          <text value="Rapport"/>
          <text variable="number" prefix="No. "/>
        </group>
```

(groups automatically suppress text values if all called variables are empty; see http://docs.citationstyles.org/en/stable/specification.html#group)

**Comment by rmzelle** [comment]
@adam3smith, do you have time to review this? I'll prep the dependent style.

**Comment by adam3smith** [comment]
@rmzelle cleaned this up some, gtg from my side now

**Comment by rmzelle** [comment]
Thanks!

**Comment by rmzelle** [comment]
(This PR was created in response to #2490)

**Pull request merged**